File: Paint_FrontEnd/src/app/Shapes/line.ts
import { IShape } from "./ishape";

export class Line extends IShape{
    private _points: number[];

    constructor(x: number, y: number, stroke: string, strokeWidth: number, fill: string, rotate: number, draggable: boolean, id: string, update: boolean, deleteflag: boolean, points: number[]){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._points = points;
        this.type = "line";
    }

    public get points(): number[] {
        return this._points;
    }
    public set points(value: number[]) {
        this._points = value;
    }
}

File: Paint_FrontEnd/src/main/java/com/paint/paint_backend/Shapes/Line.java
package com.paint.paint_backend.Shapes;

public class Line extends IShape{
    private int[] _points;
    Line(int x, int y, String stroke, double strokeWidth, String fill, int rotate, boolean draggable, String id, boolean update, boolean deleteflag, int[] points){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._points = points;
        super.set_type("line");
    }

    public int[] get_points() {
        return _points;
    }

    public void set_points(int[] _points) {
        this._points = _points;
    }
}
